src/settings.h

```diff
@@ -61,11 +64,38 @@ struct WorldOptions {
   }
 
   std::filesystem::path regionDir() {
+    size_t sepIndex;
+    std::string ns, id;
     switch (dim) {
     case NETHER:
       return std::filesystem::path(saveName) /= "DIM-1/region";
     case END:
       return std::filesystem::path(saveName) /= "DIM1/region";
+    case CUSTOM:
+      sepIndex = customDim.find_first_of(":/");
+      if (sepIndex == std::string::npos) {
+        if (customDim.substr(0, 3) == "DIM")
+          return (std::filesystem::path(saveName) /= customDim) /= "region";
+        else if (customDim == "0")
+          break;
+        else if (isNumeric(customDim.c_str()))
+          return (std::filesystem::path(saveName) /= ("DIM" + customDim)) /= "region";
+        ns = "minecraft";
+        id = customDim;
+      } else {
+        ns = customDim.substr(0, sepIndex);
+        id = customDim.substr(sepIndex + 1);
+      }
+      if (ns == "minecraft") {
+        if (id == "overworld") 
+          break;
+        else if (id == "the_nether")
+          return std::filesystem::path(saveName) /= "DIM-1/region";
+        else if (id == "the_end")
+          return std::filesystem::path(saveName) /= "DIM1/region";
+      }
```

Comment:
This code essentially does the same as the beginning of the method. Why not fully commit to namespaces from the option structure, instead of using the `Dimension` enum ?

Reply:
That's an excellent point; it seemed redundant to me too but I hadn't thought to do it. Would you like me to or are you already working on it yourself?

Is there anything else about the code that seems obscure?

Reply:
What type of custom dimensions are you referring to ? The ones added by mods or the [new 1.16.2](https://minecraft.gamepedia.com/Custom) custom ones ? Looking at your code, you mention namespaces, but use the naming schemes of the mods'.